### PR TITLE
[MM-55303] Make use of GenericStoreResult in webhook

### PR DESCRIPTION
**Pull request by Paul-Stern**
#### Summary
Make use of GenericStoreResult in channels/app/webhook.go.
Introduce a new variable to workaround IncompatibleAssign error.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/25310
Jira https://mattermost.atlassian.net/browse/MM-55303

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/webhook.go [hunk 1]
```diff
@@ -664,10 +664,10 @@ func (a *App) HandleIncomingWebhook(c request.CTX, hookID string, req *model.Inc
...
 	}
 
-	hchan := make(chan store.StoreResult, 1)
+	hchan := make(chan store.GenericStoreResult[*model.IncomingWebhook], 1)
 	go func() {
 		webhook, err := a.Srv().Store().Webhook().GetIncoming(hookID, true)
-		hchan <- store.StoreResult{Data: webhook, NErr: err}
+		hchan <- store.GenericStoreResult[*model.IncomingWebhook]{Data: webhook, NErr: err}
 		close(hchan)
 	}()
```
Comment (hanzei) [inline server/channels/app/webhook.go:672]: I wonder if we need the concurrency at all here, given that in L687 we wait for the result anyway.

**Review by hanzei — APPROVED** [review]
Nicely done :+1:

File: server/channels/app/webhook.go (hunk 1 above)
Comment (Paul-Stern) [reply server/channels/app/webhook.go:672]: A nice question I'm not ready to answer yet =)

Comment (isacikgoz) [reply server/channels/app/webhook.go:672]: Probably not a significant improvement, but we have many examples 🤷

**Review by isacikgoz — APPROVED** [review]
LGTM, thanks @Paul-Stern

**Pull request merged**